### PR TITLE
fix: Create the streams state directory if it does not exist.

### PR DESCRIPTION
**Pull request by hjafarpour**
Currently KSQL server fails if the streams state directory does not exist. With this PR, KSQL server will try to create the folder if it does not exist.
Some of the error messages were also improved.
Unit tests were updated.

### CODE REVIEW

File: ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlServerMain.java [hunk 1]
```diff
@@ -111,19 +111,37 @@ private static Executable createExecutable(
 
   @VisibleForTesting
   static void enforceStreamStateDirAvailability(final File streamsStateDir) {
-    if (!streamsStateDir.exists() || !streamsStateDir.isDirectory()) {
-      throw new KsqlServerException("The kafka streams state directory does not exist: "
-          + streamsStateDir.getPath()
-          + "\n Make sure the directory exists and is writable for KSQL server."
+    if (!streamsStateDir.exists()) {
+      final boolean mkDirSuccess = streamsStateDir.mkdir();
```
Comment (rodesai) [inline ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlServerMain.java]: Use `mkdirs` here. This way the path up to this directory will be created (this is also what streams does).

**Comment by hjafarpour** [comment]
@agavra the problem is that in KSQL we support both prefixing the streams properties and not prefixing them. I updated the PR to accept both. Please try again.

File: ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlServerMain.java [hunk 2]
```diff
@@ -99,4 +108,41 @@ private static Executable createExecutable(
...
+      throw new KsqlServerException(streamsStateDir.getPath()
+          + " is not a directory."
+          + "\n Make sure the directory exists and is writable for KSQL server "
+          + "\n or its parend directory is writbale by KSQL server"
+          + "\n or change it to a writable directory by setting '"
+          + KsqlConfig.KSQL_STREAMS_PREFIX + StreamsConfig.STATE_DIR_CONFIG
+          + "' config in the properties file."
+      );
+    }
+    if (!streamsStateDir.canWrite()) {
```
Comment (rodesai) [inline ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlServerMain.java]: On unix systems this also needs to check for execute permissions. I'm not sure what this translates to for Windows. A more robust check would be to just try creating a file and making sure it shows up in a listing.

File: ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlServerMain.java (hunk 1 above)
Comment (hjafarpour) [reply ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlServerMain.java]: Changed it to `mkdirs`.

File: ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlServerMain.java (hunk 2 above)
Comment (hjafarpour) [reply ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlServerMain.java]: Checking both writable and executable now.

**Pull request merged**